packages/media/src/createResource.ts

```diff
@@ -57,6 +62,56 @@ function createResource({
   trimData,
   needsProxy = false,
 }: ResourceInput): Resource | VideoResource | GifResource {
+  type = type || getTypeFromMime(mimeType);
+  if (type === ResourceType.VIDEO) {
```

Comment:
Hm, these two are almost identical, could we create a general resource object instead and just add the unique props inside `if` clause? E.g. `resource = { ...sharedProps }; if ( Video ) return { ...sharedProps, ... videoProps };` etc.? Would be more visible as well then which props are shared and which unique. Thoughts?

Reply:
Yeah, it could be cleaned up a bit. I'll look into it.

Reply:
I've overhauled it. PTAL.